### ISSUE TITLE
[Question] Kotlin app java.lang.LinkageError: issue ,please help

### ISSUE THREAD

**Issue by RyderChen0708** [Azure/azure-sdk-for-java#28071]
My app use microsoft.identity to login Microsoft account

but at here cause crash.
```
val interactiveBrowserCredential = InteractiveBrowserCredentialBuilder()
            .clientId(CLIENT_ID)
            .redirectUrl("")
            .build()
```
here is crash information

`java.lang.LinkageError: Package versions: jackson-annotations=0.0, jackson-core=0.0, jackson-databind=0.0, jackson-dataformat-xml=0.0, jackson-datatype-jsr310=0.0, azure-core=1.26.0, Troubleshooting version conflicts: https://aka.ms/azsdk/java/dependency/troubleshoot`

i try to modify dependencies
but not work.

```
    implementation 'com.microsoft.identity.client:msal:3.0.2'
    implementation 'com.microsoft.graph:microsoft-graph:5.19.0'
    implementation 'com.azure:azure-identity:1.4.6'
    implementation 'com.azure:azure-sdk-bom:1.2.0'
    implementation 'com.fasterxml.jackson.module:jackson-module-kotlin:2.10.5'
```

Thanks for help!

**Comment by lmolkova** [comment]
@RyderChen0708 thank you for reporting the issue.
Can you please help us understand a few more details?
1. Setup (please complete the following information):
 - OS: [e.g. iOS]
 - IDE: [e.g. IntelliJ]
 - Java version: [e.g. 8] 
 - App Server/Environment: [e.g. Tomcat, WildFly, Azure Function, Apache Spark, Databricks, IDE plugin or anything special]
 - Frameworks: [e.g. Spring Boot, Micronaut, Quarkus, etc] 

2. verbose dependency tree (`mvn dependency:tree -Dverbose`)
3. if you can share a repro app - this would be a tremendous help.

Thank you!

**Comment by jaypg** [comment]
@lmolkova Chiming in with the same issue. App is a Kotlin project and I've tried manually adding the Jackson libraries to my build.gradle but it didn't help. I've tried the interactive browser credential builder and the client credentials builder with the same crash at the same point. 

OS: Android 12
IDE: Android Studio 2021.1.1 Patch #3 (Bumblebee)
Java version: 11
App Server/Environment: Kotlin IDE plugin
Frameworks: None
